src/cljc/lab79/datomic_spec.cljc

```diff
@@ -25,19 +25,26 @@
 ;; http://docs.datomic.com/transactions.html#sec-1-2
 (s/def ::entity (s/keys :opt [:db/id]))
 
-(s/def ::tempid
-  (s/or
-    #?@(:clj [:dbid #(instance? DbId %)])
-    :string (s/and string? #(not (starts-with? % ":")))))
 (s/def :db/id
   (s/or
     :entity-id number?
-    :lookup-ref (s/tuple keyword? (s/or :string string?
-                                        :keyword keyword?
-                                        :num number?
-                                        :uuid uuid?))
-    :tempid ::tempid
-    :ident keyword?))
+    :lookup-ref ::lookup-ref
+    :tempid :db.id/tempid
+    :ident :db.id/ident))
+
+(s/def :db.id/long number?)
+(s/def ::lookup-ref
+  (s/tuple keyword?
+           (s/or :string string?
+                 :keyword keyword?
+                 :num number?
+                 :uuid uuid?)))
+(s/def :db.id/tempid
```

Comment:
We should add an alias

```clojure
(s/def ::tempid :db.id/tempid)
```

since we may be using `::tempid` in our other apps.